### DISCUSSION TITLE
Q. When I enable javaagent debug mode, I get a "io.opentelemetry.extension.trace.propagation.B3 Propagator ExtractorMultipleHeades - Invalid TraceId in B3 header:null, Returning INVALID span context" log.

### DISCUSSION THREAD

**Discussion by eottabom** [open-telemetry/opentelemetry-java-instrumentation#11618] [Q&A] [answered]
It is used as a combination of armeria and spring,
Enable and use option to change REST request to gRPC in armeria.
This problem occurs when you enable the otel java agent debug log,
`io.opentelemetry.extension.trace.propagation.B3 Propagator ExtractorMultipleHeades - Invalid TraceId in B3 header:null, Returning INVALID span context`
Is there a phenomenon like memory leak in this case?

**Comment by laurit** [comment, marked as answer]
> Is there a phenomenon like memory leak in this case?

No.
This message is logged at https://github.com/open-telemetry/opentelemetry-java/blob/0750660f6f0ab4130b81647921b21dafd48cf417/extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/B3PropagatorExtractorMultipleHeaders.java#L38 It is just an informative statement, it does not indicate that anything is wrong. It is logged for example when the request doesn't have b3 propagation headers.

**Reply by eottabom** [reply]
So if I don't want that log to come out, may I know what settings I should add?

**Reply by laurit** [reply]
Either don't enable the debug log or use `-Dio.opentelemetry.javaagent.slf4j.simpleLogger.log.io.opentelemetry.extension.trace.propagation.B3PropagatorExtractorMultipleHeaders=INFO`

**Reply by eottabom** [reply]
I don't know if it's that influence, but I don't think traceid collection is possible.

**Reply by laurit** [reply]
> I don't know if it's that influence, but I don't think traceid collection is possible.

I have no idea what you mean by this.

**Reply by eottabom** [reply]
Sorry.
I understand disabling the log, but it seems to happen because the trace id is not in the header. So, I'm trying to tell you that it's hard to trace with the trace id.

**Reply by laurit** [reply]
I'm still not sure what the question is here. We can't do anything about the trace id not being present in the header. If you want context propagation to happen then you'll have to configure the system that sends the request to add appropriate context propagation headers.

**Reply by eottabom** [reply]
When the intention of the question was grpc, there was such a problem, so I left an inquiry. Thank you for your answer. Let me check the cause further.

I'm sorry for the confusion.
